### PR TITLE
en-vi: add IWSLT'15 English-Vietnamese as new problem

### PR DESCRIPTION
**Pull request by stefan-it**
Hi,

with this PR a new problem is added: English-Vietnamese using the IWSLT'15 dataset from the [Stanford NLP group](https://nlp.stanford.edu/projects/nmt/).

I trained an English to Vietnamese model for 125k steps on a NVIDIA GTX 1060. Here are some nice comparisons of BLEU score on the `tst2013` test set. Other BLEU scores are taken from the [Towards Neural Phrase-based Machine Translation](https://arxiv.org/abs/1706.05565) paper.

| Model                                                                                                 | BLEU (Beam Search)
| ----------------------------------------------------------------------------------------------------- | ------------------
| [Luong & Manning (2015)](https://nlp.stanford.edu/pubs/luong-manning-iwslt15.pdf)                     | 23.30
| Sequence-to-sequence model with attention                                                             | 26.10
| Neural Phrase-based Machine Translation [Huang et. al. (2017)](https://arxiv.org/abs/1706.05565)      | 27.69
| Neural Phrase-based Machine Translation + LM [Huang et. al. (2017)](https://arxiv.org/abs/1706.05565) | 28.07
| Transformer (Base)                                                                                    | **28.12** (cased)
| Transformer (Base)                                                                                    | **28.97** (uncased)

### CODE REVIEW

**Comment by duyvuleo** [comment]
Hi, I trained Transformer base for this dataset with 500K steps, evaluated with t2t-bleu and got BLEU scores around 28.47 (cased) and 29.32 (uncased). But when i evaluate this result with multi-bleu.pl script from moses, I got only 27.69 (you can see my training/evaluation signatures from [here](https://github.com/duyvuleo/Transformer-DyNet/blob/master/benchmarks/iwslt-envi/)). That's weird!

It seems that t2t-bleu tends to provide higher score.

**Comment by martinpopel** [comment]
With multi-bleu.pl you need to tokenize the hypothesis and reference yourself first (and possibly normalize unicode punctuation) - depending on the way you do it you can get big difference (e.g. +- 5 BLEU). Thus multi-bleu.pl is not replicable in general.
It [seems](https://github.com/duyvuleo/Transformer-DyNet/blob/master/benchmarks/iwslt-envi/eval-envi-iwslt-tensor2tensor.sh#L34) you have forgotten to do any tokenization.
The advantage of t2t-bleu and sacrebleu is that you don't need care about tokenization - it is integrated inside.

**Comment by duyvuleo** [comment]
Ah, I forgot to mention. The decoded output from tensor2tensor for this dataset is already tokenized (look at [this](https://github.com/duyvuleo/Transformer-DyNet/blob/master/benchmarks/iwslt-envi/translation-beam5.tst2013.envi.tensor2tensor.transformer_base.translated.vi). Also, the provided reference is also already tokenized. So, the evaluation with multi-bleu.pl is fair.

**Comment by anhtuanvn** [comment]
Hi Stefan,

Thank you very much for sharing your work. 

I tried to reproduce the reported results with your pretrained model. Nevertheless, I got the error below:

> NotFoundError (see above for traceback): Restoring from checkpoint failed. This is most likely due to a Variable name or other graph key that is missing from the checkpoint. Please ensure that you have not altered the graph expected based on the checkpoint. Original error:

> Key transformer/symbol_modality_20428_512/shared/weights_0 not found in checkpoint
	 [[node save/RestoreV2_1 (defined at /usr/local/lib/python2.7/dist-packages/tensorflow_estimator/python/estimator/estimator.py:627) ]]"

Please help me to check the log file [Log.txt](https://github.com/tensorflow/tensor2tensor/files/2921999/Log.txt) and give me a piece of advice on how to solve this issue.

Thank you.

**Comment by stefan-it** [comment]
@ankushagarwal Sorry for the long response time! Could you solved the problem? What version if T2T are you using for your experiments?
